infra/src/test/kotlin/fi/fta/geoviite/infra/ui/pagemodel/map/MapToolPanelElement.kt

```diff
@@ -304,6 +304,8 @@ class SwitchCoordinatesInfoBox(by: By) : InfoBox(by) {
 class GeometryAlignmentGeneralInfoBox(by: By) : InfoBox(by) {
     fun nimi() = fieldValue("Nimi")
     fun pituusmittauslinja() = fieldValue("Pituusmittauslinja")
+
+    fun kohdistaKartalla() = clickButton("Kohdista kartalla").also { Thread.sleep(500) }
```

Comment:
Tämäkin on ihan OK, mutta voisi tutkia onko Seleniumissa mitään sisäistä tapaa tehdä tällaisia odotuksia nätimmin. Käytännössähän tässä haluttaneen odottaa, että kartta on ehtinyt kohdistua, ja painaa nappia vasta sitten.

Esim. Cypressissä kaikki UI-getterit ja kliksuttelut toimivat defaulttina niin, että niillä on timeout, ja ne koittavat timeoutin aikana sisäisesti useampaan otteeseen tehdä haluttua toimintoa. Jos toiminto onnistuu missään kohtaa timeoutin sisällä, niin sitten hypätään seuraavaan operaatioon, ja failataan operaatio jos timeoutin loppuun mennessä ei olla onnistuttu. Tällöin timeout voi olla vaikka 10sec, mutta jos operaatio on valmis jo vaikka 300ms jälkeen, niin ei oikeasti odoteta kuin jotain tuon 300ms tienoilla.

Lähinnä tämä siis sillä, että jos tässä olisi käytössä vastaava malli, niin ei tarvitsisi aina väkisellä odottaa tuota 500ms. Mutta jos tämä ei mene helpolla, niin ei mielestäni tarvi lähteä muuttamaan tässä yhteydessä. Lähinnä tykkään vaan tuosta Cypress-tyylisestä maailmasta enemmän henk.koht.

Reply:
Joo, en edes hoksannut miettiä, että voisiko tuon tehdä paremmin, copypastesin vaan jonkin toisen infoboksin vastaavasta metodista.

Ja nyt kun mietin: Kyllähän se aika paljon mälväämistä tarvitsisi että tämän saisi tehtyä oikeaoppisesti, kun pitäisi varmaan jotenkin tuossa tuolle luokalle tieto siitä, mistä geometriasta on kyse, ja sen pitäisi sitten osata katsoa, onko tämä geometria kartalla näkyvissä kokonaan. Ja jos logiikka on vaikkapa katsoa että pystyykö päätepisteitä kliksuttamaan, niin huonolla tuurilla geometria on mutkalla ja lähtötilassa päätepisteet on näkyvissä mutta jotkin pisteet välillä ei, ja sitten testi odottaa näkevänsä juuri väärän pisteen, tjms..

Reply:
Joo tämä olisi hyvä. Narkoleptiset testit on aina selkeä anti-pattern. Ne hidastaa tarpeetta ja failaa joskus randomilla jos joku muu vähän hidastelee. Ja kaupan päälle vaikea debugata kun omalla koneella juoksee nopammin.